### PR TITLE
[Enhance] Support more tensor-like operations for data structures.

### PR DESCRIPTION
**Pull request by Xiangxu-0103**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the back-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet3d/datasets/convert_utils.py [hunk 1]
```diff
@@ -330,7 +330,7 @@ def convert_annos(info: dict, cam_idx: int) -> dict:
         Box3DMode.LIDAR, np.linalg.inv(rect @ lidar2cam0), correct_yaw=True)
     # convert gt_bboxes_3d to cam coordinates
     gt_bboxes_3d = gt_bboxes_3d.convert_to(
-        Box3DMode.CAM, rect @ lidar2cami, correct_yaw=True).tensor.numpy()
```
Comment (JingweiZhang12) [inline mmdet3d/datasets/convert_utils.py:333]: Will this raise an error if gt_bboxes_3d is an instance of `BaseInstance3DBoxes`?

File: mmdet3d/datasets/transforms/loading.py [hunk 2]
```diff
@@ -387,7 +387,7 @@ def _remove_close(self,
         if isinstance(points, np.ndarray):
             points_numpy = points
         elif isinstance(points, BasePoints):
-            points_numpy = points.tensor.numpy()
```
Comment (JingweiZhang12) [inline mmdet3d/datasets/transforms/loading.py:390]: An instance of BasePoints does not have `numpy()` method ? Please check the all.

File: mmdet3d/datasets/convert_utils.py (hunk 1 above)
Comment (Xiangxu-0103) [reply mmdet3d/datasets/convert_utils.py:333]: This PR will support tensor-like methods (such as `cuda()`, `numpy()`, `detach()` etc.) for `BaseInstance3DBoxes` and `BasePoints`.

File: mmdet3d/datasets/transforms/loading.py (hunk 2 above)
Comment (Xiangxu-0103) [reply mmdet3d/datasets/transforms/loading.py:390]: The same above.

**Review by JingweiZhang12 — APPROVED** [review]

**Pull request merged**